src/components/minigraph.js

```diff
@@ -29,274 +23,283 @@ function Minigraph(props) {
+            .curve(d3.curveCardinal)
+        );
 
-    const path4 = svg4
-      .append('path')
-      .datum(data)
-      .attr('fill', 'none')
-      .attr('cursor', 'pointer')
-      .attr('stroke', '#6c757d99')
-      .attr('stroke-width', 3)
-      .attr('cursor', 'pointer')
-      .attr('cursor', 'pointer')
-      .attr(
-        'd',
-        d3
-          .line()
-          .x(function (d) {
-            return x(new Date(d['date'] + '2020'));
-          })
-          .y(function (d, i) {
-            if (i === 0) {
-              return y1(d['dailydeceased']);
-            } else {
-              return y1(d['dailydeceased']);
-            }
-          })
-          .curve(d3.curveCardinal)
-      );
+      const totalLength4 = path4.node().getTotalLength();
+      path4
+        .attr('stroke-dasharray', totalLength4 + ' ' + totalLength4)
+        .attr('stroke-dashoffset', totalLength4)
+        .transition()
+        .duration(props.animate ? 2000 : 0)
+        .attr('stroke-dashoffset', 0);
 
-    const totalLength4 = path4.node().getTotalLength();
-    path4
-      .attr('stroke-dasharray', totalLength4 + ' ' + totalLength4)
-      .attr('stroke-dashoffset', totalLength4)
-      .transition()
-      .duration(props.animate ? 2000 : 0)
-      .attr('stroke-dashoffset', 0);
+      svg4
+        .selectAll('.dot')
+        .data(data.slice(data.length - 1))
+        .enter()
+        .append('circle')
+        .attr('fill', '#6c757d')
+        .attr('stroke', '#6c757d')
+        .attr('r', 2)
+        .attr('cursor', 'pointer')
+        .attr('cx', function (d) {
+          return x(new Date(d['date'] + '2020'));
+        })
+        .attr('cy', function (d) {
+          return y1(d['dailydeceased']);
+        })
+        .on('mouseover', (d) => {
+          d3.select(d3.event.target).attr('r', '5');
+        })
+        .on('mouseout', (d) => {
+          d3.select(d3.event.target).attr('r', '2');
+        })
+        .style('opacity', 0)
+        .transition()
+        .duration(props.animate ? 3000 : 0)
+        .style('opacity', 1);
+    },
+    [width, height, svg1, props.animate, svg2, svg3, svg4]
+  );
 
-    svg4
-      .selectAll('.dot')
-      .data(data.slice(data.length - 1))
-      .enter()
-      .append('circle')
-      .attr('fill', '#6c757d')
-      .attr('stroke', '#6c757d')
-      .attr('r', 2)
-      .attr('cursor', 'pointer')
-      .attr('cx', function (d) {
-        return x(new Date(d['date'] + '2020'));
-      })
-      .attr('cy', function (d) {
-        return y1(d['dailydeceased']);
-      })
-      .on('mouseover', (d) => {
-        d3.select(d3.event.target).attr('r', '5');
-      })
-      .on('mouseout', (d) => {
-        d3.select(d3.event.target).attr('r', '2');
-      })
-      .style('opacity', 0)
-      .transition()
-      .duration(props.animate ? 3000 : 0)
-      .style('opacity', 1);
-  }
+  useEffect(() => {
+    graphData(timeseries);
+    // eslint-disable-next-line
+  }, [timeseries]);
```

Comment:
@ssonal here.

Reply:
Just patched this in #402. Should work now!

Side-note, testing these conditions is pretty painful. How are you doing it?

Reply:
Haha I feel you, it's a pain! 

So I noticed the graphs looked odd, so inspected the code and added a `console.log` in the useEffect to see how many times the state changes happened in the dependency list, and voila! The graphData was called so many times having it redrawn so many times.

Could've slipped my eyes, but hey :P

Reply:
@ssonal also I had a look at the time series graphs, I think it's the same issue. The graphs tends to be redrawn again. 

See if you can fix it too!